<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Jun Zhang">
<title>7&nbsp; Coverage plot – omicScope</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./Drug-IC50.html" rel="next">
<link href="./TME-analysis.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-edce5c54b9b59c4df3f7a2e3371b8c7a.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script defer="" data-domain="r4ds.hadley.nz" src="https://plausible.io/js/plausible.js"></script>
</head>
<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./CoveragePlot.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Coverage plot</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">omicScope</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/junjunlab/omicScope" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./BamFiles-example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Bam files example</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./RawCounts-example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Raw counts example</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./UCSCzena-example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">UCSC xena example</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./GTEX-samples-integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Integration of GTEx normal tissue samples</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./TCGAbiolinks-integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Integration of TCGAbiolinks</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./TME-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Tumor microenvironment analysis</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CoveragePlot.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Coverage plot</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Drug-IC50.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Correlation for drug IC50 and gene expression</span></span></a>
  </div>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">7.1</span> Introduction</a></li>
  <li><a href="#make-a-coverage-plot" id="toc-make-a-coverage-plot" class="nav-link" data-scroll-target="#make-a-coverage-plot"><span class="header-section-number">7.2</span> Make a coverage plot</a></li>
  <li><a href="#gene-plot" id="toc-gene-plot" class="nav-link" data-scroll-target="#gene-plot"><span class="header-section-number">7.3</span> Gene plot</a></li>
  <li><a href="#genomic-region-plot" id="toc-genomic-region-plot" class="nav-link" data-scroll-target="#genomic-region-plot"><span class="header-section-number">7.4</span> Genomic region plot</a></li>
  <li><a href="#y-axis-range-setting" id="toc-y-axis-range-setting" class="nav-link" data-scroll-target="#y-axis-range-setting"><span class="header-section-number">7.5</span> Y-axis range setting</a></li>
  <li><a href="#add-peaks-annotation" id="toc-add-peaks-annotation" class="nav-link" data-scroll-target="#add-peaks-annotation"><span class="header-section-number">7.6</span> Add peaks annotation</a></li>
  <li>
<a href="#new-features" id="toc-new-features" class="nav-link" data-scroll-target="#new-features"><span class="header-section-number">7.7</span> New features</a>
  <ul class="collapse">
<li><a href="#input-bigwig-files" id="toc-input-bigwig-files" class="nav-link" data-scroll-target="#input-bigwig-files"><span class="header-section-number">7.7.1</span> Input bigwig files</a></li>
  <li><a href="#utr-region-visualization" id="toc-utr-region-visualization" class="nav-link" data-scroll-target="#utr-region-visualization"><span class="header-section-number">7.7.2</span> UTR region visualization</a></li>
  <li><a href="#highlight-specific-region" id="toc-highlight-specific-region" class="nav-link" data-scroll-target="#highlight-specific-region"><span class="header-section-number">7.7.3</span> Highlight specific region</a></li>
  </ul>
</li>
  <li>
<a href="#implementation-with-signac" id="toc-implementation-with-signac" class="nav-link" data-scroll-target="#implementation-with-signac"><span class="header-section-number">7.8</span> Implementation with Signac</a>
  <ul class="collapse">
<li><a href="#basic-plot" id="toc-basic-plot" class="nav-link" data-scroll-target="#basic-plot"><span class="header-section-number">7.8.1</span> Basic plot</a></li>
  <li><a href="#add-links-panel" id="toc-add-links-panel" class="nav-link" data-scroll-target="#add-links-panel"><span class="header-section-number">7.8.2</span> Add links panel</a></li>
  <li><a href="#add-peaks-panel" id="toc-add-peaks-panel" class="nav-link" data-scroll-target="#add-peaks-panel"><span class="header-section-number">7.8.3</span> Add peaks panel</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/junjunlab/omicScope/edit/main/CoveragePlot.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/junjunlab/omicScope/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<h1 class="title">
<span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Coverage plot</span>
</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="introduction" class="level2" data-number="7.1"><h2 data-number="7.1" class="anchored" data-anchor-id="introduction">
<span class="header-section-number">7.1</span> Introduction</h2>
<p>The coverage_plot function in omicscope simplifies the creation of genomic coverage plots. Instead of the cumbersome process of converting BAM files to BigWig, this function works directly with your alignment files, saving you valuable time and effort. It produces crisp, publication-level graphics suitable for various sequencing applications. This makes it an essential tool not just for RNA-seq, but for anyone working with data from ATAC-seq, ChIP-seq, or CUT&amp;Tag experiments.</p>
</section><section id="make-a-coverage-plot" class="level2" data-number="7.2"><h2 data-number="7.2" class="anchored" data-anchor-id="make-a-coverage-plot">
<span class="header-section-number">7.2</span> Make a coverage plot</h2>
</section><section id="gene-plot" class="level2" data-number="7.3"><h2 data-number="7.3" class="anchored" data-anchor-id="gene-plot">
<span class="header-section-number">7.3</span> Gene plot</h2>
<p>Using aligned BAM files and a GTF annotation file, we can specify genes for visualization via the&nbsp;<code>target_gene</code>&nbsp;parameter. Here, we’ll examine the expression changes of the pluripotency gene&nbsp;<strong><em>Nanog</em></strong>&nbsp;and the housekeeping gene&nbsp;<strong><em>Actb</em></strong>&nbsp;on <strong>Day 0</strong> versus <strong>Day 10</strong> of stem cell differentiation:</p>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">omicScope</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">"../test-bam/Mus_musculus.GRCm38.102.gtf.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># test</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/0b.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10b.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day0-rep2"</span>,<span class="st">"day10-rep1"</span>,<span class="st">"day10-rep2"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Nanog"</span>,<span class="st">"Actb"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/c1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>In the bottom panel, the arrows on the gene structure indicate the <strong>direction of transcription</strong>. Arrows pointing to the right represent genes on the positive (+) strand, while those pointing to the left represent genes on the negative (-) strand.</p>
<p>To simplify the y-axis and show only a range label instead of the full axis, set both the&nbsp;<code>add_range_label</code>&nbsp;and&nbsp;<code>remove_labelY</code>&nbsp;parameters to&nbsp;<code>TRUE</code>:</p>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/0b.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10b.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day0-rep2"</span>,<span class="st">"day10-rep1"</span>,<span class="st">"day10-rep2"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Nanog"</span>,<span class="st">"Actb"</span><span class="op">)</span>,</span>
<span>              add_range_label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              remove_labelY <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/c2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Use the&nbsp;<code>sample_order</code>&nbsp;and&nbsp;<code>gene_order</code>&nbsp;parameters to reorder the samples and genes:</p>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/0b.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10b.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day0-rep2"</span>,<span class="st">"day10-rep1"</span>,<span class="st">"day10-rep2"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Nanog"</span>,<span class="st">"Actb"</span><span class="op">)</span>,</span>
<span>              sample_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day10-rep1"</span>,<span class="st">"day0-rep2"</span>,<span class="st">"day10-rep2"</span><span class="op">)</span>,</span>
<span>              gene_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Actb"</span>,<span class="st">"Nanog"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/c3.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Use the&nbsp;<code>group_name</code>&nbsp;parameter to assign samples to groups. Setting&nbsp;<code>merge_group</code>&nbsp;to&nbsp;<code>TRUE</code>&nbsp;will then display the mean coverage across biological replicates for each group:</p>
<div class="cell">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/0b.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10b.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day0-rep2"</span>,<span class="st">"day10-rep1"</span>,<span class="st">"day10-rep2"</span><span class="op">)</span>,</span>
<span>              group_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Day-0"</span>,<span class="st">"Day-0"</span>,<span class="st">"Day-10"</span>,<span class="st">"Day-10"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Nanog"</span>,<span class="st">"Actb"</span><span class="op">)</span>,</span>
<span>              merge_group <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/c4.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Set&nbsp;<code>collapse_exon = TRUE</code>&nbsp;to collapse all of a gene’s transcripts for visualization:</p>
<div class="cell">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/0b.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10b.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day0-rep2"</span>,<span class="st">"day10-rep1"</span>,<span class="st">"day10-rep2"</span><span class="op">)</span>,</span>
<span>              group_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Day-0"</span>,<span class="st">"Day-0"</span>,<span class="st">"Day-10"</span>,<span class="st">"Day-10"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Nanog"</span>,<span class="st">"Actb"</span><span class="op">)</span>,</span>
<span>              merge_group <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              collapse_exon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              exon_linewidth <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/c5.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>If you want to make the gene structure panel at the bottom shorter, you can use the&nbsp;<code>theme_sub_panel</code>&nbsp;function:</p>
<div class="cell">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/0b.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10b.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day0-rep2"</span>,<span class="st">"day10-rep1"</span>,<span class="st">"day10-rep2"</span><span class="op">)</span>,</span>
<span>              group_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Day-0"</span>,<span class="st">"Day-0"</span>,<span class="st">"Day-10"</span>,<span class="st">"Day-10"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Nanog"</span>,<span class="st">"Actb"</span><span class="op">)</span>,</span>
<span>              merge_group <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              collapse_exon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              exon_linewidth <span class="op">=</span> <span class="fl">8</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/subtheme.html">theme_sub_panel</a></span><span class="op">(</span>heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">3</span>,<span class="fl">1</span><span class="op">)</span>,<span class="st">"cm"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/c6.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Modify colors using standard&nbsp;<code>ggplot2</code>&nbsp;syntax, or set them directly with the&nbsp;<code>exon_col</code>&nbsp;(for exons) and&nbsp;<code>arrow_col</code>&nbsp;(for intron lines) parameters:</p>
<div class="cell">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/0b.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10b.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day0-rep2"</span>,<span class="st">"day10-rep1"</span>,<span class="st">"day10-rep2"</span><span class="op">)</span>,</span>
<span>              group_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Day-0"</span>,<span class="st">"Day-0"</span>,<span class="st">"Day-10"</span>,<span class="st">"Day-10"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Nanog"</span>,<span class="st">"Actb"</span><span class="op">)</span>,</span>
<span>              merge_group <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              exon_col <span class="op">=</span> <span class="st">"#05339C"</span>,arrow_col <span class="op">=</span> <span class="st">"#05339C"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>Actb <span class="op">=</span> <span class="st">"#FAB12F"</span>,Nanog <span class="op">=</span> <span class="st">"#696FC7"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>Actb <span class="op">=</span> <span class="st">"#FAB12F"</span>,Nanog <span class="op">=</span> <span class="st">"#696FC7"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/c7.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section><section id="genomic-region-plot" class="level2" data-number="7.4"><h2 data-number="7.4" class="anchored" data-anchor-id="genomic-region-plot">
<span class="header-section-number">7.4</span> Genomic region plot</h2>
<p>Besides plotting coverage for specific genes, the function also supports visualization of user-defined genomic regions. You can specify the&nbsp;<code>target_region</code>&nbsp;parameter with genomic coordinates in the format: <strong>chromosome:start_position-end_position</strong> (e.g., “chr1:1000000-2000000”):</p>
<div class="cell">
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day10-rep1"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_region <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1:127061844-128745904"</span>,</span>
<span>                                <span class="st">"2:121127504-121354559"</span><span class="op">)</span></span>
<span>              <span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/r1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>You can also set&nbsp;<code>add_gene_label = FALSE</code>&nbsp;to suppress gene name labeling:</p>
<div class="cell">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day10-rep1"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_region <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1:127061844-128745904"</span>,</span>
<span>                                <span class="st">"2:121127504-121354559"</span><span class="op">)</span>,</span>
<span>              add_range_label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              remove_labelY <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              add_gene_label <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/r2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Collapse all exons into a single track:</p>
<div class="cell">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span><span class="va">bam_file</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../test-bam/0a.sorted.bam"</span>,</span>
<span>                             <span class="st">"../test-bam/10a.sorted.bam"</span><span class="op">)</span>,</span>
<span>              <span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"day0-rep1"</span>,<span class="st">"day10-rep1"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_region <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1:127061844-128745904"</span>,</span>
<span>                                <span class="st">"2:121127504-121354559"</span><span class="op">)</span>,</span>
<span>              add_gene_label <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>              collapse_exon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              exon_linewidth <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/r3.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section><section id="y-axis-range-setting" class="level2" data-number="7.5"><h2 data-number="7.5" class="anchored" data-anchor-id="y-axis-range-setting">
<span class="header-section-number">7.5</span> Y-axis range setting</h2>
<p>The <code>coverage_plot</code> function, by default, maintains a uniform y-axis scale for all samples faceted by the same gene or genomic region. This approach ensures direct comparability, but for datasets with widely varying signal intensities across different groups, it may be necessary to define custom y-axis ranges for individual samples. This allows for a more tailored visualization that better accommodates the specific characteristics of the data.</p>
<p>This guide will explain how to set a custom y-axis display range for different samples.</p>
<p>To begin, let’s generate a default plot to observe its behavior:</p>
<div class="cell">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">omicScope</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">"../../index-data/Mus_musculus.GRCm38.102.gtf.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bam_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../5.map-data/DMSOR1Input.bam"</span>,</span>
<span>                           <span class="st">"../5.map-data/DMSOR2Input.bam"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DMSO-12h-rep1"</span>, <span class="st">"DMSO-12h-rep2"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Actb"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,</span>
<span>              add_range_label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              remove_labelY <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/rg1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>We can create a data frame containing&nbsp;<code>gene_name</code>,&nbsp;<code>y_min</code>, and&nbsp;<code>y_max</code>&nbsp;columns to specify the minimum and maximum y-axis range for each gene:</p>
<div class="cell">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">rg.cm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>gene_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Actb"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,</span>
<span>                    y_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">50</span>,<span class="op">-</span><span class="fl">25</span><span class="op">)</span>,</span>
<span>                    y_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">25</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">rg.cm</span></span>
<span><span class="co">#   gene_name y_min y_max</span></span>
<span><span class="co"># 1      Actb   -50    50</span></span>
<span><span class="co"># 2     Gapdh   -25    25</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bam_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../5.map-data/DMSOR1Input.bam"</span>,</span>
<span>                           <span class="st">"../5.map-data/DMSOR2Input.bam"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DMSO-12h-rep1"</span>, <span class="st">"DMSO-12h-rep2"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Actb"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,</span>
<span>              add_range_label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              remove_labelY <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>              set_range_val <span class="op">=</span> <span class="va">rg.cm</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/rg2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>If you want to change the y-axis range for a specific sample, you can also add a&nbsp;<code>sample</code>&nbsp;column to the data frame, specifying the range for that particular sample:</p>
<div class="cell">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">rg.cm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>gene_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Actb"</span><span class="op">)</span>,</span>
<span>                    y_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">50</span><span class="op">)</span>,</span>
<span>                    y_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span><span class="op">)</span>,</span>
<span>                    sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DMSO-12h-rep2"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bam_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../5.map-data/DMSOR1Input.bam"</span>,</span>
<span>                           <span class="st">"../5.map-data/DMSOR2Input.bam"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DMSO-12h-rep1"</span>, <span class="st">"DMSO-12h-rep2"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Actb"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,</span>
<span>              add_range_label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              remove_labelY <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>              set_range_val <span class="op">=</span> <span class="va">rg.cm</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/rg3.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Similarly, when plotting for a specific genomic region, you just need to replace the&nbsp;<code>gene_name</code>&nbsp;column with&nbsp;<code>target_region</code>:</p>
<div class="cell">
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">rg.cm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>target_region <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1:127061844-128745904"</span>,</span>
<span>                                      <span class="st">"2:121127504-121354559"</span><span class="op">)</span>,</span>
<span>                    y_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">20</span>,<span class="op">-</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>                    y_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bam_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../5.map-data/DMSOR1Input.bam"</span>,</span>
<span>                           <span class="st">"../5.map-data/DMSOR2Input.bam"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DMSO-12h-rep1"</span>, <span class="st">"DMSO-12h-rep2"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_region <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1:127061844-128745904"</span>,</span>
<span>                                <span class="st">"2:121127504-121354559"</span><span class="op">)</span>,</span>
<span>              add_range_label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              remove_labelY <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>              set_range_val <span class="op">=</span> <span class="va">rg.cm</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/rg4.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section><section id="add-peaks-annotation" class="level2" data-number="7.6"><h2 data-number="7.6" class="anchored" data-anchor-id="add-peaks-annotation">
<span class="header-section-number">7.6</span> Add peaks annotation</h2>
<p>Sometimes, you may need to add annotation regions at specific genomic locations, such as the peak regions called by MACS2 in a ChIP-seq analysis.</p>
<p>To do this, you can provide a data frame with the following columns: <code>seqnames</code>, <code>start</code>, <code>end</code>, <code>sample</code>, <code>gene_name</code>, and <code>sample_name</code>.</p>
<p>Key requirements for this data frame are:</p>
<ul>
<li><p>If you are plotting for a specific genomic interval, replace the <code>gene_name</code> column with <code>target_region</code>.</p></li>
<li><p>The <code>sample</code> column must contain the fixed string <code>"Peaks"</code> for all entries. This tells the function to draw these regions on a dedicated peak track.</p></li>
<li><p>The <code>sample_name</code> column is used to specify which original sample the peak data belongs to, allowing you to distinguish peaks from different experiments.</p></li>
</ul>
<div class="cell">
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">bd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">gtf</span>,<span class="va">gene_name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Actb"</span>, <span class="st">"Gapdh"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span> <span class="op">&amp;</span> </span>
<span>               <span class="va">transcript_biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/select.html">select</a></span><span class="op">(</span><span class="va">seqnames</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">gene_name</span>, <span class="va">transcript_id</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/mutate.html">mutate</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="st">"Peaks"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bd</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Peaks-1"</span>,<span class="st">"Peaks-2"</span>,<span class="st">"Peaks-3"</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">bd</span><span class="op">)</span>,replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">bd</span><span class="op">)</span></span>
<span><span class="co">#        seqnames     start       end gene_name      transcript_id sample sample_name</span></span>
<span><span class="co"># 641049        5 142906652 142906754      Actb ENSMUST00000100497  Peaks     Peaks-1</span></span>
<span><span class="co"># 641050        5 142905564 142905692      Actb ENSMUST00000100497  Peaks     Peaks-3</span></span>
<span><span class="co"># 641053        5 142905237 142905476      Actb ENSMUST00000100497  Peaks     Peaks-2</span></span>
<span><span class="co"># 641055        5 142904344 142904782      Actb ENSMUST00000100497  Peaks     Peaks-2</span></span>
<span><span class="co"># 641057        5 142904067 142904248      Actb ENSMUST00000100497  Peaks     Peaks-2</span></span>
<span><span class="co"># 641059        5 142903115 142903941      Actb ENSMUST00000100497  Peaks     Peaks-1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, let’s generate the plot by passing the&nbsp;<code>peaks</code>&nbsp;data frame we created above to the&nbsp;<code>peaks_data</code>&nbsp;argument:</p>
<div class="cell">
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bam_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../5.map-data/DMSOR1Input.bam"</span>,</span>
<span>                           <span class="st">"../5.map-data/DMSOR2Input.bam"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DMSO-12h-rep1"</span>, <span class="st">"DMSO-12h-rep2"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Actb"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,</span>
<span>              add_range_label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              remove_labelY <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>              peaks_data <span class="op">=</span> <span class="va">bd</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/rg5.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section><section id="new-features" class="level2" data-number="7.7"><h2 data-number="7.7" class="anchored" data-anchor-id="new-features">
<span class="header-section-number">7.7</span> New features</h2>
<p>We are constantly working to enhance the&nbsp;<code>coverage_plot</code>&nbsp;function. Below is a description of the latest features and improvements.</p>
<section id="input-bigwig-files" class="level3" data-number="7.7.1"><h3 data-number="7.7.1" class="anchored" data-anchor-id="input-bigwig-files">
<span class="header-section-number">7.7.1</span> Input bigwig files</h3>
<p>In addition to directly processing <strong>BAM</strong> files,&nbsp;<code>coverage_plot</code>&nbsp;now supports <strong>BigWig</strong> files. As a popular format for visualization with a smaller file size, BigWig provides a lightweight alternative for generating coverage plots.</p>
<div class="cell">
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">omicScope</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">"../../index-data/Mus_musculus.GRCm38.102.gtf.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bw_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../7.bw-data/scrinput.bw"</span>,<span class="st">"../7.bw-data/scrDinput.bw"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dmso"</span>,<span class="st">"treat"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="st">"Nanog"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/bwp.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="540"></p>
</figure>
</div>
<p>Plotting a specific genomic region:</p>
<div class="cell">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bw_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../7.bw-data/scrinput.bw"</span>,<span class="st">"../7.bw-data/scrDinput.bw"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dmso"</span>,<span class="st">"treat"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_region <span class="op">=</span> <span class="st">"5:76176615-76376575"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/bwp2.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="540"></p>
</figure>
</div>
</section><section id="utr-region-visualization" class="level3" data-number="7.7.2"><h3 data-number="7.7.2" class="anchored" data-anchor-id="utr-region-visualization">
<span class="header-section-number">7.7.2</span> UTR region visualization</h3>
<p>As you may notice in the plot above, <strong>UTR</strong> regions are rendered with thinner lines than <strong>CDS</strong> regions. This new feature allows for better visual distinction between the coding and non-coding portions of a gene. To render all exons with a uniform width, you can set&nbsp;<code>show_utr = FALSE</code>:</p>
<div class="cell">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bw_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"../7.bw-data/scrinput.bw"</span>,<span class="st">"../7.bw-data/scrDinput.bw"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dmso"</span>,<span class="st">"treat"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="st">"Nanog"</span>,</span>
<span>              show_utr <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/bwp3.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="540"></p>
</figure>
</div>
</section><section id="highlight-specific-region" class="level3" data-number="7.7.3"><h3 data-number="7.7.3" class="anchored" data-anchor-id="highlight-specific-region">
<span class="header-section-number">7.7.3</span> Highlight specific region</h3>
<p>Sometimes, you may want to visually emphasize specific genomic areas. To do this, simply prepare a data frame containing the coordinates of your target regions and pass it to the <code>highlight_region</code> parameter.</p>
<p>In the example below, we’ll highlight several exons within the <strong>Nanog</strong> gene:</p>
<div class="cell">
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/filter.html">filter</a></span><span class="op">(</span><span class="va">gene_name</span> <span class="op">==</span> <span class="st">"Nanog"</span> <span class="op">&amp;</span> <span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span> <span class="op">&amp;</span> <span class="va">transcript_id</span> <span class="op">==</span> <span class="st">"ENSMUST00000012540"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/arrange.html">arrange</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/desc.html">desc</a></span><span class="op">(</span><span class="va">width</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/select.html">select</a></span><span class="op">(</span><span class="va">seqnames</span>,<span class="va">start</span>,<span class="va">end</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/mutate.html">mutate</a></span><span class="op">(</span>type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/context.html">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">g</span></span>
<span><span class="co">#   seqnames     start       end type</span></span>
<span><span class="co"># 1        6 122713142 122714633    1</span></span>
<span><span class="co"># 2        6 122707568 122707933    2</span></span>
<span><span class="co"># 3        6 122711538 122711803    3</span></span>
<span><span class="co"># 4        6 122712629 122712715    4</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot highlight region:</p>
<div class="cell">
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bam_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"scrinput.bam"</span>,<span class="st">"scrinput.bam"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dmso"</span>,<span class="st">"treat"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="st">"Nanog"</span>,</span>
<span>              highlight_region <span class="op">=</span> <span class="va">g</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/hl1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="751"></p>
</figure>
</div>
<p>The name of the column to use for color mapping:</p>
<div class="cell">
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/omicScope/man/coverage_plot.html">coverage_plot</a></span><span class="op">(</span>bam_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"scrinput.bam"</span>,<span class="st">"scrDinput.bam"</span><span class="op">)</span>,</span>
<span>              sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dmso"</span>,<span class="st">"treat"</span><span class="op">)</span>,</span>
<span>              gtf_file <span class="op">=</span> <span class="va">gtf</span>,</span>
<span>              target_gene <span class="op">=</span> <span class="st">"Nanog"</span>,</span>
<span>              highlight_region <span class="op">=</span> <span class="va">g</span>,</span>
<span>              highlight_col_aes <span class="op">=</span> <span class="st">"type"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/hl2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section></section><section id="implementation-with-signac" class="level2" data-number="7.8"><h2 data-number="7.8" class="anchored" data-anchor-id="implementation-with-signac">
<span class="header-section-number">7.8</span> Implementation with Signac</h2>
<p>When analyzing single-cell epigenomic data such as scATAC-seq, packages like Signac have become a standard tool in the field. Building on its powerful foundation, we have developed an enhanced alternative to Signac’s <code>CoveragePlot</code> function. Our implementation, which generates plots like the <code>coverage_plot</code> shown previously, offers improved customizations and features.</p>
<p>To begin, we assume you have a <code>SeuratObject</code> containing scATAC-seq data. The workflow for creating this object is detailed in the official <a href="https://stuartlab.org/signac/articles/pbmc_vignette">“Analyzing PBMC scATAC-seq”</a> tutorial, so we will not repeat it here. Let’s start by loading the required libraries and our sample data:</p>
<div class="cell">
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">omicScope</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/stuart-lab/signac">Signac</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/single-cell-utils.R"</span>, package <span class="op">=</span> <span class="st">"omicScope"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load PBMC dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"pbmc.rda"</span><span class="op">)</span></span>
<span><span class="va">pbmc</span></span>
<span><span class="co"># An object of class Seurat </span></span>
<span><span class="co"># 185049 features across 9641 samples within 2 assays </span></span>
<span><span class="co"># Active assay: peaks (165434 features, 165434 variable features)</span></span>
<span><span class="co"># 2 layers present: counts, data</span></span>
<span><span class="co"># 1 other assay present: RNA</span></span>
<span><span class="co"># 2 dimensional reductions calculated: lsi, umap</span></span>
<span></span>
<span><span class="co"># load gtf file</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/GFFFile-class.html">import.gff</a></span><span class="op">(</span><span class="st">"gencode.v49.annotation.gtf.gz"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="basic-plot" class="level3" data-number="7.8.1"><h3 data-number="7.8.1" class="anchored" data-anchor-id="basic-plot">
<span class="header-section-number">7.8.1</span> Basic plot</h3>
<p>Let’s first visualize the data using Signac’s default plotting function:</p>
<div class="cell">
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/Signac/man/CoveragePlot.html">CoveragePlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc</span>,</span>
<span>             region <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD8A"</span>,<span class="st">"CD4"</span><span class="op">)</span>,</span>
<span>             extend.upstream <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>             extend.downstream <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/s1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Now, let’s make a plot using&nbsp;<code>scCoverage_plot</code>:</p>
<div class="cell">
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">scCoverage_plot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc</span>,</span>
<span>                gtf_file <span class="op">=</span> <span class="va">hm</span>,</span>
<span>                target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD8A"</span>,<span class="st">"CD4"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/sc1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Alternatively, you can specify a genomic region to plot:</p>
<div class="cell">
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">scCoverage_plot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc</span>,</span>
<span>                gtf_file <span class="op">=</span> <span class="va">hm</span>,</span>
<span>                target_region <span class="op">=</span> <span class="st">"chr2:86784605-86808396"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/sc2.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="603"></p>
</figure>
</div>
<p>Adding a new signal range label:</p>
<div class="cell">
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">scCoverage_plot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc</span>,</span>
<span>                gtf_file <span class="op">=</span> <span class="va">hm</span>,</span>
<span>                target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD4"</span>,<span class="st">"LYZ"</span><span class="op">)</span>,</span>
<span>                add_range_label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                remove_labelY <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                range_pos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.9</span>, <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>panel.spacing.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0</span>,<span class="st">"mm"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/sc3.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Visualizing multiple genes by collapsing different transcripts into a single gene model:</p>
<div class="cell">
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">scCoverage_plot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc</span>,</span>
<span>                gtf_file <span class="op">=</span> <span class="va">hm</span>,</span>
<span>                target_gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'MS4A1'</span>, <span class="st">'CD3D'</span>, <span class="st">'LEF1'</span>,<span class="st">'TREM1'</span><span class="op">)</span>,</span>
<span>                add_range_label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                remove_labelY <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                range_pos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.9</span>, <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>                collapse_exon <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>panel.spacing.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0</span>,<span class="st">"mm"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/sc4.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="2783"></p>
</figure>
</div>
</section><section id="add-links-panel" class="level3" data-number="7.8.2"><h3 data-number="7.8.2" class="anchored" data-anchor-id="add-links-panel">
<span class="header-section-number">7.8.2</span> Add links panel</h3>
<p>Link data is typically used for analyzing datasets such as Hi-C, co-accessibility, and enhancer-promoter interactions. For this example, we will randomly generate links between different sites within the CD8A gene for demonstration purposes. Please note that this data is purely illustrative and has no actual biological significance.</p>
<p><strong>Important:</strong>&nbsp;When generating the&nbsp;<code>links_data</code>, the&nbsp;<code>seqnames</code>,&nbsp;<code>start</code>,&nbsp;<code>end</code>, and&nbsp;<code>sample</code>&nbsp;columns are required. Furthermore, if you are using the&nbsp;<code>target_gene</code>&nbsp;parameter, a corresponding&nbsp;<code>gene_name</code>&nbsp;column must be provided. Similarly, using the&nbsp;<code>target_region</code>&nbsp;parameter requires a&nbsp;<code>target_region</code>&nbsp;column:</p>
<div class="cell">
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">gfl</span> <span class="op">&lt;-</span> <span class="va">hm</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/filter.html">filter</a></span><span class="op">(</span><span class="va">gene_name</span> <span class="op">==</span> <span class="st">"CD8A"</span> <span class="op">&amp;</span> <span class="va">type</span> <span class="op">==</span> <span class="st">"gene"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">from</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="va">gfl</span><span class="op">$</span><span class="va">start</span><span class="op">:</span><span class="va">gfl</span><span class="op">$</span><span class="va">end</span>,<span class="fl">20</span>,replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="va">links</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">gfl</span><span class="op">$</span><span class="va">seqnames</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                    start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="va">gfl</span><span class="op">$</span><span class="va">start</span><span class="op">:</span><span class="va">gfl</span><span class="op">$</span><span class="va">end</span>,<span class="fl">40</span>,replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>                    end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="va">gfl</span><span class="op">$</span><span class="va">start</span><span class="op">:</span><span class="va">gfl</span><span class="op">$</span><span class="va">end</span>,<span class="fl">40</span>,replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>                    score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.01</span><span class="op">)</span>,<span class="fl">40</span>,replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>                    sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"links-1"</span>,<span class="st">"links-2"</span><span class="op">)</span>,<span class="fl">20</span><span class="op">)</span>,</span>
<span>                    gene_name <span class="op">=</span> <span class="st">"CD8A"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/filter.html">filter</a></span><span class="op">(</span><span class="va">end</span> <span class="op">&gt;</span> <span class="va">start</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">links</span><span class="op">)</span></span>
<span><span class="co">#   seqnames    start      end score  sample gene_name</span></span>
<span><span class="co"># 1     chr2 86794131 86797385  0.56 links-1      CD8A</span></span>
<span><span class="co"># 2     chr2 86795722 86803883  0.08 links-1      CD8A</span></span>
<span><span class="co"># 3     chr2 86790482 86793071  0.35 links-2      CD8A</span></span>
<span><span class="co"># 4     chr2 86804095 86808098  0.99 links-1      CD8A</span></span>
<span><span class="co"># 5     chr2 86797230 86798150  0.92 links-2      CD8A</span></span>
<span><span class="co"># 6     chr2 86786291 86799074  0.65 links-1      CD8A</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>To visualize the links, pass the data you prepared to the&nbsp;<code>links_data</code>&nbsp;argument:</p>
<div class="cell">
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">scCoverage_plot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc</span>,</span>
<span>                gtf_file <span class="op">=</span> <span class="va">hm</span>,</span>
<span>                target_gene <span class="op">=</span> <span class="st">"CD8A"</span>,</span>
<span>                links_data <span class="op">=</span> <span class="va">links</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/sc5.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section><section id="add-peaks-panel" class="level3" data-number="7.8.3"><h3 data-number="7.8.3" class="anchored" data-anchor-id="add-peaks-panel">
<span class="header-section-number">7.8.3</span> Add peaks panel</h3>
<p>Peaks data generally refers to DNA binding regions found by peak callers such as MACS2, although it can represent any genomic regions.</p>
<p>For data preparation, your data frame must include the following columns: <code>seqnames</code>, <code>start</code>, <code>end</code>, and <code>sample</code>. Critically, every entry in the <code>sample</code> column must be the string <code>'Peaks'</code>. Additionally, you need a <code>sample_name</code> column to identify different samples. Please also note the conditional requirements: if you are using the <code>target_gene</code> parameter, a corresponding <code>gene_name</code> column must be provided. Similarly, using the <code>target_region</code> parameter requires a <code>target_region</code> column.</p>
<p>Below, we will generate some random BED-formatted data to visualize this:</p>
<div class="cell">
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">gfl</span><span class="op">$</span><span class="va">seqnames</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                    start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="va">gfl</span><span class="op">$</span><span class="va">start</span><span class="op">:</span><span class="va">gfl</span><span class="op">$</span><span class="va">end</span>,<span class="fl">40</span>,replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>                    end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="va">gfl</span><span class="op">$</span><span class="va">start</span><span class="op">:</span><span class="va">gfl</span><span class="op">$</span><span class="va">end</span>,<span class="fl">40</span>,replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>                    score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.01</span><span class="op">)</span>,<span class="fl">40</span>,replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>                    sample <span class="op">=</span> <span class="st">"Peaks"</span>,</span>
<span>                    sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"peaks-1"</span>,<span class="st">"peaks-2"</span><span class="op">)</span>,<span class="fl">20</span><span class="op">)</span>,</span>
<span>                    gene_name <span class="op">=</span> <span class="st">"CD8A"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/filter.html">filter</a></span><span class="op">(</span><span class="va">end</span> <span class="op">&gt;</span> <span class="va">start</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">)</span></span>
<span><span class="co">#   seqnames    start      end score sample sample_name gene_name</span></span>
<span><span class="co"># 1     chr2 86787803 86797842  0.42  Peaks     peaks-2      CD8A</span></span>
<span><span class="co"># 2     chr2 86787870 86793469  0.72  Peaks     peaks-1      CD8A</span></span>
<span><span class="co"># 3     chr2 86790426 86805815  0.92  Peaks     peaks-2      CD8A</span></span>
<span><span class="co"># 4     chr2 86787409 86792736  0.02  Peaks     peaks-1      CD8A</span></span>
<span><span class="co"># 5     chr2 86793777 86797484  1.00  Peaks     peaks-2      CD8A</span></span>
<span><span class="co"># 6     chr2 86793063 86802052  0.29  Peaks     peaks-1      CD8A</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>To visualize the peaks, pass your data to the&nbsp;<code>peaks_data</code>&nbsp;argument:</p>
<div class="cell">
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">scCoverage_plot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc</span>,</span>
<span>                gtf_file <span class="op">=</span> <span class="va">hm</span>,</span>
<span>                target_gene <span class="op">=</span> <span class="st">"CD8A"</span>,</span>
<span>                peaks_data <span class="op">=</span> <span class="va">peaks</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/sc6.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Next, let’s visualize both the links and peaks data on the same plot:</p>
<div class="cell">
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">scCoverage_plot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc</span>,</span>
<span>                gtf_file <span class="op">=</span> <span class="va">hm</span>,</span>
<span>                target_gene <span class="op">=</span> <span class="st">"CD8A"</span>,</span>
<span>                links_data <span class="op">=</span> <span class="va">links</span>,</span>
<span>                peaks_data <span class="op">=</span> <span class="va">peaks</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/sc7.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>


</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/github\.com\/junjunlab\/omicScope");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./TME-analysis.html" class="pagination-link" aria-label="Tumor microenvironment analysis">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Tumor microenvironment analysis</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./Drug-IC50.html" class="pagination-link" aria-label="Correlation for drug IC50 and gene expression">
        <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Correlation for drug IC50 and gene expression</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>omicScope documentation was written by Jun Zhang.</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/junjunlab/omicScope/edit/main/CoveragePlot.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/junjunlab/omicScope/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


</body></html>