<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Jun Zhang">
<title>6&nbsp; Tumor microenvironment analysis â€“ omicScope</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./TCGAbiolinks-integration.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-f1075ac1703e165f0c9b3df650605ffb.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script defer="" data-domain="r4ds.hadley.nz" src="https://plausible.io/js/plausible.js"></script>
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./TME-analysis.html"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Tumor microenvironment analysis</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">omicScope</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/junjunlab/omicScope" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./BamFiles-example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Bam files example</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./RawCounts-example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Raw counts example</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./UCSCzena-example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">UCSC xena example</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./GTEX-samples-integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Integration of GTEx normal tissue samples</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./TCGAbiolinks-integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Integration of TCGAbiolinks</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./TME-analysis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Tumor microenvironment analysis</span></span></a>
  </div>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">6.1</span> Introduction</a></li>
  <li><a href="#tme-analysis" id="toc-tme-analysis" class="nav-link" data-scroll-target="#tme-analysis"><span class="header-section-number">6.2</span> TME analysis</a></li>
  <li>
<a href="#visualization" id="toc-visualization" class="nav-link" data-scroll-target="#visualization"><span class="header-section-number">6.3</span> Visualization</a>
  <ul class="collapse">
<li><a href="#boxplot" id="toc-boxplot" class="nav-link" data-scroll-target="#boxplot"><span class="header-section-number">6.3.1</span> Boxplot</a></li>
  <li><a href="#heatmap-plot" id="toc-heatmap-plot" class="nav-link" data-scroll-target="#heatmap-plot"><span class="header-section-number">6.3.2</span> Heatmap plot</a></li>
  <li><a href="#signature-results-plot" id="toc-signature-results-plot" class="nav-link" data-scroll-target="#signature-results-plot"><span class="header-section-number">6.3.3</span> Signature results plot</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/junjunlab/omicScope/edit/main/TME-analysis.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/junjunlab/omicScope/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<h1 class="title">
<span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Tumor microenvironment analysis</span>
</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="introduction" class="level2" data-number="6.1"><h2 data-number="6.1" class="anchored" data-anchor-id="introduction">
<span class="header-section-number">6.1</span> Introduction</h2>
<p>omicScope integrates the IOBR package to facilitate comprehensive <strong>tumor microenvironment (TME)</strong> analysis. The&nbsp;<code>run_TME_analysis</code>&nbsp;function enables one-click TME characterization, storing all results in the&nbsp;<code>tmeData</code>&nbsp;slot. Users have the flexibility to select their preferred analytical strategy: either signature-based scoring using curated gene sets to evaluate TME characteristics, or deconvolution algorithms to estimate the abundance and composition of immune cell types within the tumor.</p>
</section><section id="tme-analysis" class="level2" data-number="6.2"><h2 data-number="6.2" class="anchored" data-anchor-id="tme-analysis">
<span class="header-section-number">6.2</span> TME analysis</h2>
<p>Use the&nbsp;<code>method</code>&nbsp;parameter to choose between two analytical strategies:&nbsp;<code>"signature"</code>&nbsp;for signature-based scoring or&nbsp;<code>"deconvo"</code>&nbsp;for deconvolution analysis. Then specify the exact algorithm using either&nbsp;<code>deconvo_method</code>&nbsp;(for deconvolution) or&nbsp;<code>signature_method</code>&nbsp;(for signature analysis). Before proceeding, you need to install the <strong>IOBR</strong> package from its official website:</p>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">run_TME_analysis</span><span class="op">(</span><span class="va">obj</span>,</span>
<span>                        method <span class="op">=</span> <span class="st">"deconvo"</span>,</span>
<span>                        deconvo_method <span class="op">=</span> <span class="st">"cibersort"</span><span class="op">)</span></span>
<span><span class="co"># ==========================================================================</span></span>
<span><span class="co">#   IOBR v0.99.99  Immuno-Oncology Biological Research </span></span>
<span><span class="co">#   For Tutorial: https://iobr.github.io/book/</span></span>
<span><span class="co">#   For Help: https://github.com/IOBR/IOBR/issues</span></span>
<span><span class="co"># </span></span>
<span><span class="co">#  If you use IOBR in published research, please cite:</span></span>
<span><span class="co">#  DQ Zeng, YR Fang, ..., GC Yu*, WJ Liao*, </span></span>
<span><span class="co">#  Enhancing immuno-oncology investigations through multidimensional decoding </span></span>
<span><span class="co">#  of tumor microenvironment with IOBR 2.0. Cell Rep Methods 4, 100910 (2024). </span></span>
<span><span class="co">#  &amp;  </span></span>
<span><span class="co">#  YR Fang, ..., WJ Liao*, DQ Zeng*, </span></span>
<span><span class="co">#  Systematic Investigation of Tumor Microenvironment and </span></span>
<span><span class="co">#  Antitumor Immunity With IOBR, Med Research (2025). </span></span>
<span><span class="co">#  https://onlinelibrary.wiley.com/doi/epdf/10.1002/mdr2.70001 </span></span>
<span><span class="co"># ==========================================================================</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># &gt;&gt;&gt; Running CIBERSORT</span></span>
<span> </span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">run_TME_analysis</span><span class="op">(</span><span class="va">obj</span>,</span>
<span>                        method <span class="op">=</span> <span class="st">"signature"</span>,</span>
<span>                        signature_method <span class="op">=</span> <span class="st">"pca"</span><span class="op">)</span></span>
<span><span class="co"># &gt;&gt;&gt; Calculating signature score using PCA method</span></span>
<span><span class="co"># &gt;&gt;&gt; log2 transformation was finished</span></span>
<span><span class="co"># There were 50 or more warnings (use warnings() to see the first 50)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Check the results:</p>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">tmeData</span><span class="op">)</span></span>
<span><span class="co"># Formal class 'SimpleList' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">#   ..@ listData       :List of 2</span></span>
<span><span class="co">#   .. ..$ deconvoRes  : tibble [600 Ã— 26] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#   .. .. ..$ ID                                    : chr [1:600] "TCGA-44-6147-01B-06R-A277-07" "TCGA-44-6147-01A-11R-1755-07" "TCGA-44-7661-01A-11R-2066-07" "TCGA-05-4396-01A-21R-1858-07" ...</span></span>
<span><span class="co">#   .. .. ..$ B_cells_naive_CIBERSORT               : num [1:600] 0.099 0.1708 0.0512 0.0333 0.0206 ...</span></span>
<span><span class="co">#   .. .. ..$ B_cells_memory_CIBERSORT              : num [1:600] 0.0817 0 0 0 0 ...</span></span>
<span><span class="co">#   .. .. ..$ Plasma_cells_CIBERSORT                : num [1:600] 0.12177 0.22735 0.00504 0.08953 0.07076 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cells_CD8_CIBERSORT                 : num [1:600] 0.0166 0.0788 0.0678 0.0947 0.1116 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cells_CD4_naive_CIBERSORT           : num [1:600] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cells_CD4_memory_resting_CIBERSORT  : num [1:600] 0.363 0.132 0.138 0.176 0.102 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cells_CD4_memory_activated_CIBERSORT: num [1:600] 0.03211 0.02895 0.12342 0.00512 0.02324 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cells_follicular_helper_CIBERSORT   : num [1:600] 0 0 0.0208 0.0593 0.0591 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cells_regulatory_(Tregs)_CIBERSORT  : num [1:600] 0 0.0255 0.0572 0.0833 0.0218 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cells_gamma_delta_CIBERSORT         : num [1:600] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#   .. .. ..$ NK_cells_resting_CIBERSORT            : num [1:600] 0 0.0271 0 0.0225 0.0189 ...</span></span>
<span><span class="co">#   .. .. ..$ NK_cells_activated_CIBERSORT          : num [1:600] 0.00647 0 0.01206 0 0.01423 ...</span></span>
<span><span class="co">#   .. .. ..$ Monocytes_CIBERSORT                   : num [1:600] 0.00051 0 0.05242 0.00818 0.01099 ...</span></span>
<span><span class="co">#   .. .. ..$ Macrophages_M0_CIBERSORT              : num [1:600] 0 0.1267 0.0166 0.115 0.1048 ...</span></span>
<span><span class="co">#   .. .. ..$ Macrophages_M1_CIBERSORT              : num [1:600] 0.0283 0.042 0.0682 0.0646 0.0865 ...</span></span>
<span><span class="co">#   .. .. ..$ Macrophages_M2_CIBERSORT              : num [1:600] 0.108 0.079 0.154 0.177 0.254 ...</span></span>
<span><span class="co">#   .. .. ..$ Dendritic_cells_resting_CIBERSORT     : num [1:600] 0 0 0.0628 0.0254 0.0288 ...</span></span>
<span><span class="co">#   .. .. ..$ Dendritic_cells_activated_CIBERSORT   : num [1:600] 0.06385 0.00959 0.09873 0.00743 0 ...</span></span>
<span><span class="co">#   .. .. ..$ Mast_cells_resting_CIBERSORT          : num [1:600] 0.0374 0.0508 0.0573 0.0143 0.0268 ...</span></span>
<span><span class="co">#   .. .. ..$ Mast_cells_activated_CIBERSORT        : num [1:600] 0 0 0 0.00257 0 ...</span></span>
<span><span class="co">#   .. .. ..$ Eosinophils_CIBERSORT                 : num [1:600] 0.032467 0 0.000917 0 0.014565 ...</span></span>
<span><span class="co">#   .. .. ..$ Neutrophils_CIBERSORT                 : num [1:600] 0.00945 0.00137 0.01276 0.02122 0.03073 ...</span></span>
<span><span class="co">#   .. .. ..$ P-value_CIBERSORT                     : num [1:600] 0.003 0.002 0.013 0.033 0.048 ...</span></span>
<span><span class="co">#   .. .. ..$ Correlation_CIBERSORT                 : num [1:600] 0.393 0.486 0.255 0.186 0.149 ...</span></span>
<span><span class="co">#   .. .. ..$ RMSE_CIBERSORT                        : num [1:600] 0.921 0.873 0.975 1.01 1.022 ...</span></span>
<span><span class="co">#   .. ..$ signatureRes: tibble [600 Ã— 312] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#   .. .. ..$ ID                                                 : chr [1:600] "TCGA-44-6147-01B-06R-A277-07" "TCGA-44-6147-01A-11R-1755-07" "TCGA-44-7661-01A-11R-2066-07" "TCGA-05-4396-01A-21R-1858-07" ...</span></span>
<span><span class="co">#   .. .. ..$ CD_8_T_effector                                    : num [1:600] 0.123 -0.688 1.775 -1.586 2.836 ...</span></span>
<span><span class="co">#   .. .. ..$ DDR                                                : num [1:600] 2.482 -0.432 10.376 2.417 -2.956 ...</span></span>
<span><span class="co">#   .. .. ..$ APM                                                : num [1:600] -3.942 0.219 2.672 1.062 2.835 ...</span></span>
<span><span class="co">#   .. .. ..$ Immune_Checkpoint                                  : num [1:600] 1.031 0.619 3.723 -2.192 2.547 ...</span></span>
<span><span class="co">#   .. .. ..$ CellCycle_Reg                                      : num [1:600] 4.9797 -0.0903 1.0643 2.2662 -1.8748 ...</span></span>
<span><span class="co">#   .. .. ..$ Pan_F_TBRs                                         : num [1:600] -1.354 2.299 4.014 -0.713 2.682 ...</span></span>
<span><span class="co">#   .. .. ..$ EMT1                                               : num [1:600] -4.944 0.568 -2.085 1.285 -0.507 ...</span></span>
<span><span class="co">#   .. .. ..$ EMT2                                               : num [1:600] -1.049 0.636 3.633 -1.162 2.579 ...</span></span>
<span><span class="co">#   .. .. ..$ EMT3                                               : num [1:600] 3.224 0.664 2.018 1.005 -1.301 ...</span></span>
<span><span class="co">#   .. .. ..$ WNT_target                                         : num [1:600] 0.485 0.253 1.934 0.541 -0.203 ...</span></span>
<span><span class="co">#   .. .. ..$ FGFR3_related                                      : num [1:600] -1.359 1.214 1.87 -0.41 -0.217 ...</span></span>
<span><span class="co">#   .. .. ..$ Cell_cycle                                         : num [1:600] -4.62 -2.38 9.46 1.03 2.73 ...</span></span>
<span><span class="co">#   .. .. ..$ Mismatch_Repair                                    : num [1:600] -4.6338 -0.7052 4.6153 0.7524 0.0886 ...</span></span>
<span><span class="co">#   .. .. ..$ Homologous_recombination                           : num [1:600] -0.416 -0.786 5.158 -0.994 0.869 ...</span></span>
<span><span class="co">#   .. .. ..$ Nucleotide_excision_repair                         : num [1:600] -5.429 -0.628 5.428 1.23 0.488 ...</span></span>
<span><span class="co">#   .. .. ..$ DNA_replication                                    : num [1:600] -6.751 -1.466 6.203 -0.328 1.789 ...</span></span>
<span><span class="co">#   .. .. ..$ Base_excision_repair                               : num [1:600] -8.43 -0.45 4.212 -0.488 0.774 ...</span></span>
<span><span class="co">#   .. .. ..$ TMEscoreA_CIR                                      : num [1:600] -0.264 -0.885 7.293 -3.855 4.498 ...</span></span>
<span><span class="co">#   .. .. ..$ TMEscoreB_CIR                                      : num [1:600] 12.36 8.96 8.02 -2.46 -2.53 ...</span></span>
<span><span class="co">#   .. .. ..$ CD8_Rooney_et_al                                   : num [1:600] -1.108 -0.568 1.032 -2.207 1.822 ...</span></span>
<span><span class="co">#   .. .. ..$ B_cells_Rooney_et_al                               : num [1:600] 6.169 4.564 0.609 -2.129 -1.05 ...</span></span>
<span><span class="co">#   .. .. ..$ Treg_Rooney_et_al                                  : num [1:600] -0.291 0.378 2.764 -0.714 0.439 ...</span></span>
<span><span class="co">#   .. .. ..$ Macrophages_Rooney_et_al                           : num [1:600] -1.941 -0.959 0.823 -1.71 1.651 ...</span></span>
<span><span class="co">#   .. .. ..$ Neutrophils_Rooney_et_al                           : num [1:600] 1.011 0.6 2.635 -0.428 -0.448 ...</span></span>
<span><span class="co">#   .. .. ..$ pDCs_Rooney_et_al                                  : num [1:600] -1.629 0.865 5.736 -1.557 0.786 ...</span></span>
<span><span class="co">#   .. .. ..$ MHC_Class_I_Rooney_et_al                           : num [1:600] -3.3054 -0.0326 1.7711 1.0391 2.6542 ...</span></span>
<span><span class="co">#   .. .. ..$ Co_stimulation_APC_Rooney_et_al                    : num [1:600] 2.036 0.502 2.503 -1.299 2.316 ...</span></span>
<span><span class="co">#   .. .. ..$ Co_stimulation_T_cell_Rooney_et_al                 : num [1:600] 2.805 0.877 3.632 -1.309 0.11 ...</span></span>
<span><span class="co">#   .. .. ..$ Co_inhibition_APC_Rooney_et_al                     : num [1:600] -0.000842 -0.946441 2.964509 -1.559692 2.018563 ...</span></span>
<span><span class="co">#   .. .. ..$ Co_inhibition_T_cell_Rooney_et_al                  : num [1:600] 3.43 0.31 3.44 -2.34 1.68 ...</span></span>
<span><span class="co">#   .. .. ..$ Type_I_IFN_Reponse_Rooney_et_al                    : num [1:600] 0.684 2.265 3.26 -1.011 1.516 ...</span></span>
<span><span class="co">#   .. .. ..$ Type_II_IFN_Reponse_Rooney_et_al                   : num [1:600] 0.807 0.924 2.485 -0.539 -1.23 ...</span></span>
<span><span class="co">#   .. .. ..$ MHC_Class_I                                        : num [1:600] -4.243 0.272 3.069 1.01 2.815 ...</span></span>
<span><span class="co">#   .. .. ..$ MHC_Class_II                                       : num [1:600] -2.874 0.439 2.309 -1.171 2.261 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cell_inflamed_GEP_Ayers_et_al                    : num [1:600] -0.607 0.863 2.99 -2.768 2.37 ...</span></span>
<span><span class="co">#   .. .. ..$ IFNG_signature_Ayers_et_al                         : num [1:600] 1.498 -0.218 2 -0.82 2.533 ...</span></span>
<span><span class="co">#   .. .. ..$ MDSC_Wang_et_al                                    : num [1:600] -0.2 0.831 6.026 -4.219 2.782 ...</span></span>
<span><span class="co">#   .. .. ..$ GPAGs                                              : num [1:600] 0.888 1.241 3.668 -0.808 -0.607 ...</span></span>
<span><span class="co">#   .. .. ..$ PPAGs                                              : num [1:600] -2.034 1.09 6.784 0.794 1.342 ...</span></span>
<span><span class="co">#   .. .. ..$ HLA_signature_gene                                 : num [1:600] 8.87 6.67 13.22 -12.7 4.15 ...</span></span>
<span><span class="co">#   .. .. ..$ PPARgama_target_genes                              : num [1:600] 0.909 -0.113 1.065 3.033 -0.849 ...</span></span>
<span><span class="co">#   .. .. ..$ B_cells_Danaher_et_al                              : num [1:600] 3.6503 4.2358 0.0501 -2.3161 -0.415 ...</span></span>
<span><span class="co">#   .. .. ..$ Cytotoxic_cells_Danaher_et_al                      : num [1:600] -1.942 -1.063 0.676 -2.331 1.239 ...</span></span>
<span><span class="co">#   .. .. ..$ DC_Danaher_et_al                                   : num [1:600] -1.208 -0.432 3.057 -1.287 3.067 ...</span></span>
<span><span class="co">#   .. .. ..$ Exhausted_CD8_Danaher_et_al                        : num [1:600] 0.117 -0.383 1.32 -1.712 1.414 ...</span></span>
<span><span class="co">#   .. .. ..$ Macrophages_Danaher_et_al                          : num [1:600] 1.122 -0.321 1.983 -1.764 1.854 ...</span></span>
<span><span class="co">#   .. .. ..$ Mast_cells_Danaher_et_al                           : num [1:600] 1.215 1.811 0.838 -1.362 -1.911 ...</span></span>
<span><span class="co">#   .. .. ..$ Neutrophils_Danaher_et_al                          : num [1:600] -0.316 0.391 1.247 -1.944 0.574 ...</span></span>
<span><span class="co">#   .. .. ..$ NK_CD56dim_cells_Danaher_et_al                     : num [1:600] -0.9082 -0.498 -0.2173 -0.1352 0.0911 ...</span></span>
<span><span class="co">#   .. .. ..$ NK_cells_Danaher_et_al                             : num [1:600] -1.421 -0.755 0.318 -1.2 1.274 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cells_Danaher_et_al                              : num [1:600] 2.044 1.08 1.959 -1.498 0.545 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Release_of_cancer_cell_antigens                : num [1:600] -8.341 0.148 2.668 1.781 0.76 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Cancer_antigen_presentation_1                  : num [1:600] 3.852 -0.416 3.284 -1.918 0.969 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Cancer_antigen_presentation_2                  : num [1:600] 1.3181 -0.5345 0.0936 -0.5675 -0.2321 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Priming_and_activation_1                       : num [1:600] 3.81 1.86 6.21 -3.82 2.57 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Priming_and_activation_2                       : num [1:600] 2.04 1.01 4.23 -2.5 1.89 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Trafficking_of_immune_cells_to_tumors          : num [1:600] -0.358 1.631 4.721 -3.331 2.953 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Infiltration_of_immune_cells_into_tumors_1     : num [1:600] -0.0437 0.963 -0.6779 -0.5759 -0.2505 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Infiltration_of_immune_cells_into_tumors_2     : num [1:600] -1.308 -0.548 1.67 -0.357 -0.167 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Recognition_of_cancer_cells_by_T_cells_1       : num [1:600] 2.524 0.949 5.37 -1.598 3.069 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Recognition_of_cancer_cells_by_T_cells_2       : num [1:600] 3.394 0.625 3.38 -1.598 1.546 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Killing_of_cancer_cells_1                      : num [1:600] 0.103 -0.94 1.561 -1.144 1.34 ...</span></span>
<span><span class="co">#   .. .. ..$ TIP_Killing_of_cancer_cells_2                      : num [1:600] -0.336 1.176 2.645 -2.732 2.793 ...</span></span>
<span><span class="co">#   .. .. ..$ TLS_Nature                                         : num [1:600] 2.85 1.84 2.52 -2.18 -1.6 ...</span></span>
<span><span class="co">#   .. .. ..$ TMEscoreA_plus                                     : num [1:600] -0.783 -1.412 3.924 -2.298 3.51 ...</span></span>
<span><span class="co">#   .. .. ..$ TMEscoreB_plus                                     : num [1:600] 3.13 3.03 3.98 -1.54 -1.29 ...</span></span>
<span><span class="co">#   .. .. ..$ B_cells_Bindea_et_al                               : num [1:600] 4.008 5.212 -0.699 -2.735 -0.45 ...</span></span>
<span><span class="co">#   .. .. ..$ T_cells_Bindea_et_al                               : num [1:600] 3.899 1.477 3.308 -2.232 0.332 ...</span></span>
<span><span class="co">#   .. .. ..$ T_helper_cells_Bindea_et_al                        : num [1:600] 3.038 0.334 3.402 2.985 -2.312 ...</span></span>
<span><span class="co">#   .. .. ..$ Tcm_Bindea_et_al                                   : num [1:600] 12.88 3.09 2.92 2.28 -4.83 ...</span></span>
<span><span class="co">#   .. .. ..$ Tem_Bindea_et_al                                   : num [1:600] 2.1243 0.8415 2.816 -0.0614 -1.5163 ...</span></span>
<span><span class="co">#   .. .. ..$ Th1_cells_Bindea_et_al                             : num [1:600] 2.42 0.18 4.66 -2.26 2 ...</span></span>
<span><span class="co">#   .. .. ..$ Th2_cells_Bindea_et_al                             : num [1:600] -0.707 -1.509 3.473 -0.233 1.319 ...</span></span>
<span><span class="co">#   .. .. ..$ TFH_Bindea_et_al                                   : num [1:600] 1.11 2.47 1.15 1.36 -3.09 ...</span></span>
<span><span class="co">#   .. .. ..$ Th17_cells_Bindea_et_al                            : num [1:600] -1.5419 0.1849 -0.0592 1.1036 -1.9045 ...</span></span>
<span><span class="co">#   .. .. ..$ CD8_T_cells_Bindea_et_al                           : num [1:600] 4.963 1.885 2.288 0.442 -1.62 ...</span></span>
<span><span class="co">#   .. .. ..$ Tgd_Bindea_et_al                                   : num [1:600] 2.8243 -0.0955 0.1329 -0.4503 -0.327 ...</span></span>
<span><span class="co">#   .. .. ..$ Cytotoxic_cells_Bindea_et_al                       : num [1:600] -0.559 -0.291 0.754 -2.63 0.587 ...</span></span>
<span><span class="co">#   .. .. ..$ NK_cells_Bindea_et_al                              : num [1:600] -1.044 2.327 2.476 -0.394 -1.232 ...</span></span>
<span><span class="co">#   .. .. ..$ NK_CD56dim_cells_Bindea_et_al                      : num [1:600] -1.221 -0.902 0.372 -0.406 0.836 ...</span></span>
<span><span class="co">#   .. .. ..$ DC_Bindea_et_al                                    : num [1:600] -2.076 0.192 4.468 -1.535 1.697 ...</span></span>
<span><span class="co">#   .. .. ..$ iDC_Bindea_et_al                                   : num [1:600] -2.909 -0.266 3.38 -2.546 0.688 ...</span></span>
<span><span class="co">#   .. .. ..$ aDC_Bindea_et_al                                   : num [1:600] -1.839 0.408 1.592 -0.107 1.584 ...</span></span>
<span><span class="co">#   .. .. ..$ Eosinophils_Bindea_et_al                           : num [1:600] 4.224 2.183 2.505 -0.203 -3.138 ...</span></span>
<span><span class="co">#   .. .. ..$ Macrophages_Bindea_et_al                           : num [1:600] 0.343 -0.223 3.779 -3.36 3.431 ...</span></span>
<span><span class="co">#   .. .. ..$ Mast_cells_Bindea_et_al                            : num [1:600] 2.24 3.05 1.89 -1.65 -2.97 ...</span></span>
<span><span class="co">#   .. .. ..$ Neutrophils_Bindea_et_al                           : num [1:600] 1.343 2.033 3.831 -2.884 0.491 ...</span></span>
<span><span class="co">#   .. .. ..$ SW480_cancer_cells_Bindea_et_al                    : num [1:600] -7.919 0.753 3.743 -0.803 2.096 ...</span></span>
<span><span class="co">#   .. .. ..$ Normal_mucosa_Bindea_et_al                         : num [1:600] 3.088 4.14 3.96 -0.244 1.043 ...</span></span>
<span><span class="co">#   .. .. ..$ Antigen_Processing_and_Presentation_Li_et_al       : num [1:600] -7.73 0.55 6.48 -1.66 3.72 ...</span></span>
<span><span class="co">#   .. .. ..$ Antimicrobials_Li_et_al                            : num [1:600] 3.24 2.28 13.11 -5.94 2.68 ...</span></span>
<span><span class="co">#   .. .. ..$ BCR_Signaling_Pathway_Li_et_al                     : num [1:600] 5.178 3.933 5.504 -0.601 -0.67 ...</span></span>
<span><span class="co">#   .. .. ..$ Chemokines_Li_et_al                                : num [1:600] -1.032 0.653 7.175 -3.592 3.14 ...</span></span>
<span><span class="co">#   .. .. ..$ Chemokine_Receptors_Li_et_al                       : num [1:600] 2.64283 2.15404 5.87339 -2.14288 -0.00636 ...</span></span>
<span><span class="co">#   .. .. ..$ Cytokines_Li_et_al                                 : num [1:600] 6.4835 3.864 8.2362 -2.4767 -0.0478 ...</span></span>
<span><span class="co">#   .. .. ..$ Cytokine_Receptors_Li_et_al                        : num [1:600] 6.5 6.17 9.29 -1.19 -3.56 ...</span></span>
<span><span class="co">#   .. .. ..$ Interferons_Li_et_al                               : num [1:600] 3.5293 -0.5651 0.0918 0.0784 -0.6349 ...</span></span>
<span><span class="co">#   .. .. ..$ Interferon_Receptor_Li_et_al                       : num [1:600] 1.882 1.034 0.513 -1.277 -0.536 ...</span></span>
<span><span class="co">#   .. .. .. [list output truncated]</span></span>
<span><span class="co">#   ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#   ..@ elementMetadata: NULL</span></span>
<span><span class="co">#   ..@ metadata       : list()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="visualization" class="level2" data-number="6.3"><h2 data-number="6.3" class="anchored" data-anchor-id="visualization">
<span class="header-section-number">6.3</span> Visualization</h2>
<p>Weâ€™ve included the&nbsp;<code>tme_plot</code>&nbsp;function to generate visualizations of your TME analysis.</p>
<section id="boxplot" class="level3" data-number="6.3.1"><h3 data-number="6.3.1" class="anchored" data-anchor-id="boxplot">
<span class="header-section-number">6.3.1</span> Boxplot</h3>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">tme_plot</span><span class="op">(</span><span class="va">obj</span>, </span>
<span>         method <span class="op">=</span> <span class="st">"deconvo"</span>,</span>
<span>         type <span class="op">=</span> <span class="st">"box"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/tme-box.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section><section id="heatmap-plot" class="level3" data-number="6.3.2"><h3 data-number="6.3.2" class="anchored" data-anchor-id="heatmap-plot">
<span class="header-section-number">6.3.2</span> Heatmap plot</h3>
<div class="cell">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">tme_plot</span><span class="op">(</span><span class="va">obj</span>, </span>
<span>         method <span class="op">=</span> <span class="st">"deconvo"</span>,</span>
<span>         type <span class="op">=</span> <span class="st">"heatmap"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/tme-heatmap.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section><section id="signature-results-plot" class="level3" data-number="6.3.3"><h3 data-number="6.3.3" class="anchored" data-anchor-id="signature-results-plot">
<span class="header-section-number">6.3.3</span> Signature results plot</h3>
<div class="cell">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">tme_plot</span><span class="op">(</span><span class="va">obj</span>, </span>
<span>         method <span class="op">=</span> <span class="st">"signature"</span>,</span>
<span>         type <span class="op">=</span> <span class="st">"heatmap"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="image/tme-sig.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>


</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/github\.com\/junjunlab\/omicScope");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./TCGAbiolinks-integration.html" class="pagination-link" aria-label="Integration of TCGAbiolinks">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Integration of TCGAbiolinks</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>omicScope documentation was written by Jun Zhang.</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/junjunlab/omicScope/edit/main/TME-analysis.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/junjunlab/omicScope/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


</body></html>